pkg/engine/ahocorasickcore.go

```diff
@@ -60,29 +62,44 @@ func NewAhoCorasickCore(allDetectors []detectors.Detector) *AhoCorasickCore {
 	}
 }
 
-// MatchString performs a string match using the Aho-Corasick algorithm, returning an array of matches.
-// Designed for internal use within the AhoCorasickCore component.
-func (ac *AhoCorasickCore) MatchString(input string) []*ahocorasick.Match {
-	return ac.prefilter.MatchString(strings.ToLower(input))
-}
-
-// PopulateDetectorsByMatch populates the given detectorMap based on the Aho-Corasick match results.
-// This method is designed to reuse the same map for performance optimization,
-// reducing the need for repeated allocations within each detector worker in the engine.
-func (ac *AhoCorasickCore) PopulateDetectorsByMatch(match *ahocorasick.Match, detectors map[detectorspb.DetectorType]detectors.Detector) bool {
+// populateDetectorsByMatch populates the given detectorMap based on the Aho-Corasick match results.
+// This method populates an existing map rather than allocating a new one to avoid allocating a new
+// map for every match.
+func (ac *AhoCorasickCore) populateDetectorsByMatch(match *ahocorasick.Match, detectors map[DetectorKey]detectors.Detector) bool {
 	matchedDetectorKeys, ok := ac.keywordsToDetectors[match.MatchString()]
 	if !ok {
 		return false
 	}
 	for _, key := range matchedDetectorKeys {
-		detectors[key.detectorType] = ac.detectorsByKey[key]
+		detectors[key] = ac.detectorsByKey[key]
 	}
 	return true
```

Comment:
Oh, I think I did this because the return value is ignored at the call site so I was just like eh, return whatever. But if that's the reason I should just remove the return value entirely.

Reply:
Oh, with your changes it looks like the return value is being used to assign `gotAny` in `PopulateMatchingDetectors`.

It's not clear to me what the return value of this function is indicating though.

Reply:
Yeah, but the return value of `PopulateMatchingDetectors` is ignored at its only call site. This was the original code:
```
for _, match := range e.ahoCorasickCore.MatchString(string(decoded.Chunk.Data)) {
    if !e.ahoCorasickCore.PopulateDetectorsByMatch(match, chunkSpecificDetectors) {
        continue
    }
}
```